addons/point_of_sale/models/pos_category.py

```diff
@@ -48,3 +48,10 @@ def _unlink_except_session_open(self):
     def _compute_has_image(self):
         for category in self:
             category.has_image = bool(category.image_128)
+
+    def _get_descendants(self, available_categories):
+        available_categories |= self
+        for child in self.child_id:
+            available_categories |= child
+            available_categories |= child._get_descendants(available_categories)
+        return available_categories
```

Comment:
i don't really get this. I would think that self would be a recordset of categories and we want to return a recordset of all the categories that are children of at least one of the categories in self. This way in the `_get_available_categories` that you create in `pos.config` you could do:
```py
def _get_available_categories(self):
    return self.iface_available_categ_ids._get_descendants()
```